frontend/herdForm.tsx

```diff
@@ -3,115 +3,288 @@
+            setNew(false)
+          }
         },
         error => console.error(error)
       );
     }
-  }, [props])
+    setLoading(false);
+  }, [id])
 
+  /**
+   * Sets a single key `label` in the `herd` form to `value` (if herd isn't
+   * undefined).
+   */
   const setFormField = <K extends keyof Herd>(label: K, value: Herd[K]) => {
     herd && setHerd({...herd, [label]: value})
   }
 
+  /**
+   * Returns a genebank, identified by `genebankId`, as a React-Select option.
+   * @param genebankId
+   */
+  const genebankOption = (genebankId: number) => {
+    const genebank = genebanks.find((g: Genebank) => g.id == herd.genebank)
+    if (genebank) {
+      return {value: genebank.id, label: genebank.name}
+    }
+    return null
+  }
+
+  /**
+   * sends a POST request to create a new herd in the database if the `isNew` is
+   * `true`, otherwise sends am UPDATE request to update a current database
+   * entry. This function will also update the `genebanks` context with the
+   * new information. If a new entry is created, the page will reroute to edit
+   * that entry.
+   */
   const submitForm = () => {
     if (herd == undefined) {
       return
     }
     const postData: Herd = {...herd};
     delete postData["individuals"];
-    update('/api/manage/herd', postData).then(
-      data => {
-        switch (data.status) {
-          case "updated": console.debug("updated successfully"); break;
-          default: console.debug("status:", data)// "failed" or other erro
+    postData["physical_address"] = `${postData["physical_address"]}|${postalcode}|${postalcity}`
+
+    if (isNew) {
+      createHerd(postData).then(
+        status => {
+          if (status == 'success') {
+            const genebank = genebanks.find((g: Genebank) => g.id == postData.genebank)
+            if (genebank) {
+              genebank.herds.push(postData)
+              setGenebanks(Object.assign([], genebanks))
+
+              // navigate to new herd to allow continued editing
+              history.push(`/manage/${genebank?.name}/${postData.herd}`)
+            }
+          }
         }
-      },
-      error => console.error(error)
-    )
+      )
+    } else {
+      updateHerd(postData).then(
+        status => {
+          if (status == 'updated') {
+            const genebank = genebanks.find((g: Genebank) => g.id == postData.genebank)
+            if (genebank) {
+              let toUpdate = genebank.herds.find((h: Herd) => h.herd == id)
+              if (toUpdate) {
+                toUpdate.herd_name = postData.herd_name
+                setGenebanks(Object.assign([], genebanks))
+              }
+            }
+          }
+        }
+      );
+    }
   }
 
   return <>
-    {loading && <h2>Loading...</h2> ||  herd &&
+    {loading && <h2>Loading...</h2> ||
       <>
-        <h1>{herd.name || `Besättning ${herd.id}`}</h1>
-        <form className={classes.form} noValidate autoComplete="off">
-          {simpleFields.map((field, i) => {
```

Comment:
simpleFields seemed like a reasonable abstraction here

Reply:
It did - but since the next step will be to start adding validation and (probably) permission levels I'll keep them separate until I know what the final form will look like. I agree that this is a very verbose form though!